src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -46,6 +46,7 @@ PreReleaseRequest<I>::~PreReleaseRequest() {
   if (!m_shutting_down) {
     m_image_ctx.state->handle_prepare_lock_complete();
   }
+  delete m_on_finish;
```

Comment:
@joscollin m_on_finish is destroyed in  `PreReleaseRequest<I>::finish()`, when `m_on_finish->complete()` is called.
See jenkins failures. I believe these are due to m_on_finish double free.